server/index.js

```diff
@@ -1,41 +1,37 @@
-import path from "path";
-import { fileURLToPath } from "url";
 import express from "express";
-import bodyParser from "body-parser";
+import cors from "cors";
 import dotenv from "dotenv";
 import router from "./routes.js";
-import cors from "cors";
+import path from "path";
+
 import { cleanReturnPayload } from "./utils/cleanReturnPayload.js";
-dotenv.config();
+import { fileURLToPath } from "url";
+dotenv.config({ path: "../.env" });
 
-const PORT = process.env.PORT || 3001;
-const app = express();
+function checkEnvVariables() {
+  const requiredEnvVariables = ["INTERACTIVE_KEY", "INTERACTIVE_SECRET"];
+  const missingVariables = requiredEnvVariables.filter((variable) => !process.env[variable]);
 
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
+  if (missingVariables.length > 0) {
+    throw new Error(`Missing required environment variables in the .env file: ${missingVariables.join(", ")}`);
+  } else {
+    console.log("All required environment variables provided.");
+  }
+}
+checkEnvVariables();
 
-app.use(function (req, res, next) {
-  const ogSend = res.send;
-  res.send = function (data) {
-    if (data) {
-      try {
-        const cleanData = cleanReturnPayload(typeof data === "string" ? JSON.parse(data) : data, "topia");
-        res.send = ogSend;
-        return res.send(cleanData);
-      } catch (error) {
-        console.error(error);
-        next();
-      }
-    }
-  };
-  next();
-});
+const PORT = process.env.PORT || 3000;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
 
 app.use("/api", router);
 
 if (process.env.NODE_ENV === "development") {
   const corsOptions = {
-    origin: "http://localhost:3000",
+    origin: ["http://localhost:3000", "http://localhost:5173"],
```

Comment:
No need for the second element here because we can assume the client always runs in port 3001. Also I think the first element should be `http://localhost:3001`